src/test/java/com/jcabi/log/VerboseProcessTest.java

```diff
@@ -477,4 +481,125 @@ public int decide(final LoggingEvent event) {
             return decision;
         }
     }
+
+    /**
+     * Runs a java process which will throw a stack trace and makes sure it 
+     * will group the stack trace.
+     */
+    @Test
+    public void logCompleteStackTrace() {
+        final org.apache.log4j.Logger rootLogger = 
+                org.apache.log4j.Logger.getRootLogger();
+        final TestAppender appender = new TestAppender();
+        rootLogger.addAppender(appender);
+        final String[] commands = { getJavaExecLoc(), ("-cp"), 
+                System.getProperty("java.class.path"),
+                "com.jcabi.log.VerboseProcessExample" };
+        final ProcessBuilder builder = new ProcessBuilder(commands);
+        VerboseProcess process = null;
+        RuntimeException thrownException = null;
+        try {
+            process = new VerboseProcess(builder, Level.INFO, Level.SEVERE);
+            process.stdout();
+        } catch (final RuntimeException e) {
+            thrownException = e;
+        } finally {
+            rootLogger.removeAppender(appender);
+            if (process != null) {
+                process.close();
+            }
+        }
+        Assert.assertNotNull(thrownException);
+        MatcherAssert.assertThat(thrownException.getMessage(),
+                Matchers.containsString(VerboseProcessExample.SYSOUT_1));
+        MatcherAssert.assertThat(thrownException.getMessage(),
+                Matchers.containsString(VerboseProcessExample.SYSOUT_2));
+        MatcherAssert.assertThat(thrownException.getMessage(),        
+        Matchers.containsString(VerboseProcessExample.SYSOUT_3));
+        MatcherAssert.assertThat(thrownException.getMessage(),
+                Matchers.containsString(VerboseProcessExample.SYSOUT_4));
+        boolean foundCompleteStack = false;
+        for (final LoggingEvent event : appender.getLog()) {
+            final String message = (String) event.getMessage();
+            if (message.contains(VerboseProcessExample.THROWN_ERR_MSG)) {
+                final boolean containsCaughtException =
+                        message.contains(VerboseProcessExample.CAUGHT_ERR_MSG);
+                Assert.assertTrue(containsCaughtException);
+                foundCompleteStack = containsCaughtException;
+            }
+        }
+        Assert.assertTrue(foundCompleteStack);
+    }
+
+    /**
+     * Gets the location of Java, whether on Linux of Windows.
+     * @return
+     */
+    public static String getJavaExecLoc() {
+        String javaExec = null;
+        String sunLibPath = System.getProperty("java.home");
+        javaExec = new String();
+        if (SystemUtils.IS_OS_WINDOWS) {
+            // Windows
+            try {
+                final String sunLibPathNew =
```

Comment:
@dean-e-clark again, please read http://www.yegor256.com/2015/01/12/compound-name-is-code-smell.html